Lib/test/test_tokenize.py

```diff
@@ -1937,25 +1937,34 @@ def test_string(self):
     """)
 
         self.check_tokenize('f"abc"', """\
-    STRING     'f"abc"'      (1, 0) (1, 6)
+    FSTRING_START 'f"'          (1, 0) (1, 2)
+    FSTRING_END 'abc'         (1, 2) (1, 5)
     """)
 
         self.check_tokenize('fR"a{b}c"', """\
-    STRING     'fR"a{b}c"'   (1, 0) (1, 9)
+    FSTRING_START 'fR"'         (1, 0) (1, 3)
+    FSTRING_MIDDLE 'a'           (1, 3) (1, 4)
+    LBRACE     '{'           (1, 4) (1, 5)
+    NAME       'b'           (1, 5) (1, 6)
+    RBRACE     '}'           (1, 6) (1, 7)
+    FSTRING_END 'c'           (1, 7) (1, 8)
     """)
 
         self.check_tokenize('f"""abc"""', """\
-    STRING     'f\"\"\"abc\"\"\"'  (1, 0) (1, 10)
+    FSTRING_START 'f\"""'        (1, 0) (1, 4)
+    FSTRING_END 'abc'         (1, 4) (1, 7)
     """)
 
         self.check_tokenize(r'f"abc\
 def"', """\
-    STRING     'f"abc\\\\\\ndef"' (1, 0) (2, 4)
+    FSTRING_START \'f"\'          (1, 0) (1, 2)
+    FSTRING_END 'abc\\\\\\ndef'  (2, -1) (2, 3)
```

Comment:
This may be pointing to a bug in the implementation as the column should not be negative. We may be handling incorrectly the column offset when there are backslashes

Reply:
CC @isidentical as you worked with that part last time IIRC

Reply:
The line number and the end column/line seems correct but that -1 is weird as fuck

Reply:
Yep, I think that needs updating. @mgmacias95 can you update that? You need to update the `lineno` and `start_line` variables based on this:

https://github.com/pablogsal/cpython/blob/95b5d07c8296c175cb9a0a695b460a42ec901567/Parser/tokenizer.c#L1566

Basically, before we only took into account `STRING` tokens and now we need to take into account all the new ones. Btw, try to make a macro because `(type == FSTRING_MIDDLE || type == FSTRING_END || ...)` inlined sucks 😅

Reply:
@mgmacias95 for your records: `Parser/tokenizer.c` is the tokenizer implementation and `Python/Python-tokenize.c` is an extension module that exposes the tokenizer inplementation to Python for testing. This is exposed as the `_tokenize` module:

```
>>> import _tokenize
>>> _tokenize
<module '_tokenize' (built-i
```

Basically `Python/Python-tokenize.c` marshalls the C objects into Python objects and apparently also introduces some bugs as a feature free of charge 😆

Reply:
solved in #51